### PR TITLE
Added documentation for identifier

### PR DESCRIPTION
**Pull request by XAMPPRocky**

### CODE REVIEW

File: asn1-core/src/identifier.rs [hunk 1]
```diff
@@ -1,3 +1,96 @@
+//! This module encompasses the representation of the ASN.1 tags.
+//!
+//! Prior to the introduction of the `AUTOMATIC TAGS` construct, ASN.1
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]: Comma may be unnecessary

File: asn1-core/src/identifier.rs [hunk 2]
```diff
@@ -1,3 +1,96 @@
...
+//! `CHOICE` in a later version, add them to the end.
+//!
+//! `Universal` class tags are used only in X.680 and their use is reserved for
+//! that specification to define ASN.1 data types.
+//!
+//! # Example
+//! A frequently encountered style for the use of tags is to assign an
+//! `Application` class tag precisely once in the entire specification, using it
+//! to identify a type that finds wide, scattered, use within the specification.
+//! An `Application` class tag is also frequently used (once only) to tag the
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]: > frequently used (once only)

Perhaps a little confusing, does it mean once per module?

File: asn1-core/src/identifier.rs [hunk 3]
```diff
@@ -7,11 +100,15 @@ pub enum Class {
 }
 
 /// An abstract representation of the identifier octets used in BER, CER, and
-/// DER to identify .
+/// DER to identify a encoded value.
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]:
```suggestion
/// DER to identify an encoded value.
```

File: asn1-core/src/identifier.rs [hunk 4]
```diff
@@ -1,3 +1,96 @@
...
+//!
+//! # Example
+//! Textual use of `IMPLICIT` with every tag is generally found only in older
+//! specifications. BER produces a less compact representation when explicit
+//! tagging is used than when implicit tagging is used. PER produces the same
+//! compact encoding in both cases. With BER and explicit tagging, there is more
+//! visibility of the underlying type (`INTEGER`, `REAL`, `BOOLEAN`, etc.) in
+//! the encoded data. These guidelines use implicit tagging in the examples
+//! whenever it is legal to do so. This may, depending on the encoding rules,
+//! result in a compact representation, which is highly desirable in some
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]:
```suggestion
//! result in a compact representation which is highly desirable in some
```

File: asn1-core/src/identifier.rs [hunk 5]
```diff
@@ -1,3 +1,96 @@
...
+//! applications. In other applications, compactness may be less important than,
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]:
```suggestion
//! applications. In other applications compactness may be less important than,
```

File: asn1-core/src/identifier.rs [hunk 6]
```diff
@@ -1,3 +1,96 @@
...
+//! for example, the ability to carry out strong type-checking. In the latter
+//! case, explicit tagging can be used.
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]:
```suggestion
//! case explicit tagging can be used.
```

File: asn1-core/src/identifier.rs [hunk 7]
```diff
@@ -7,11 +100,15 @@ pub enum Class {
 }
 
 /// An abstract representation of the identifier octets used in BER, CER, and
-/// DER to identify .
+/// DER to identify a encoded value.
+///
 #[derive(Debug, Clone, Copy, Eq, PartialEq)]
 pub struct Identifier {
+    /// The class of the identifier.
```
Comment (chocol4te) [inline asn1-core/src/identifier.rs]:
```suggestion
    /// The class of the Identifier.
```

File: asn1-core/src/identifier.rs (hunk 2 above)
Comment (XAMPPRocky) [reply asn1-core/src/identifier.rs]: _"A frequently encountered style for the use of tags is to assign an `Application` class tag precisely once in the entire specification"_

Comment (chocol4te) [reply asn1-core/src/identifier.rs]: Would _"frequently (still only only once per specification)"_ be better?

Or maybe

_"An `Application` class tag is also used to tag the"_

**Pull request closed without merging**